crates/node-core/src/node_config.rs

```diff
@@ -552,16 +533,14 @@ impl Default for NodeConfig {
     }
 }
 
-/// Returns the address if this is an [`Ipv4Addr`].
-pub fn ipv4(ip: IpAddr) -> Option<Ipv4Addr> {
+fn ipv4(ip: IpAddr) -> Option<Ipv4Addr> {
     match ip {
         IpAddr::V4(ip) => Some(ip),
         IpAddr::V6(_) => None,
     }
 }
 
-/// Returns the address if this is an [`Ipv6Addr`].
-pub fn ipv6(ip: IpAddr) -> Option<Ipv6Addr> {
+fn ipv6(ip: IpAddr) -> Option<Ipv6Addr> {
```

Comment:
hmm, didn't know this is on main as well, really don't get why IpAddr doesn't have a function for this.

meh the std library does not have this for some reason.

looks like we only use this one, can we remove this function and use match instead?